src/test/java/com/google/sps/DatastorePlaceGuideRepositoryTest.java

```diff
@@ -0,0 +1,207 @@
+    assertEquals(IS_PUBLIC, placeGuide.getProperty(
+                                            DatastorePlaceGuideRepository.IS_PUBLIC_PROPERTY));
+    assertEquals(LENGTH, placeGuide.getProperty(DatastorePlaceGuideRepository.LENGTH_PROPERTY));
+    assertEquals(DESC, placeGuide.getProperty(DatastorePlaceGuideRepository.DESC_PROPERTY));
+    assertEquals(IMG_KEY, placeGuide.getProperty(DatastorePlaceGuideRepository.IMG_KEY_PROPERTY));
+  }
+
+  @Test
+  public void getAllPlaceGuides_placeGuideExists_returnPlaceGuide() {
+    Entity placeGuideEntity = new Entity(DatastorePlaceGuideRepository.ENTITY_KIND, ID);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.NAME_PROPERTY, NAME);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.AUDIO_KEY_PROPERTY, AUDIO_KEY);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.CREATOR_ID_PROPERTY, CREATOR_ID);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.PLACE_ID_PROPERTY, PLACE_ID);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.IS_PUBLIC_PROPERTY, IS_PUBLIC);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.COORD_PROPERTY, COORD);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.DESC_PROPERTY, DESC);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.LENGTH_PROPERTY, LENGTH);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.IMG_KEY_PROPERTY, IMG_KEY);
+    datastore.put(placeGuideEntity);
+    List<PlaceGuide> results = placeGuideRepository.getAllPlaceGuides();
+    assertEquals(1, results.size());
+    PlaceGuide result = results.get(0);
+    assertTrue(testPlaceGuide.equals(result));
+  }
+
+  @Test
+  public void getAllPlaceGuides_placeGuideDoesntExists_placeGuideListIsEmpty() {
+    List<PlaceGuide> results = placeGuideRepository.getCreatedPlaceGuides(CREATOR_ID);
+    assertTrue(results.isEmpty());
+  }
+
+  @Test
+  public void getCreatedPlaceGuides_placeGuideExistsButNotOwnedByUser_placeGuideListIsEmpty() {
+    Entity placeGuideEntity = new Entity(DatastorePlaceGuideRepository.ENTITY_KIND, ID);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.NAME_PROPERTY, NAME);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.AUDIO_KEY_PROPERTY, AUDIO_KEY);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.CREATOR_ID_PROPERTY, CREATOR_ID);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.PLACE_ID_PROPERTY, PLACE_ID);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.IS_PUBLIC_PROPERTY, IS_PUBLIC);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.COORD_PROPERTY, COORD);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.DESC_PROPERTY, DESC);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.LENGTH_PROPERTY, LENGTH);
+    placeGuideEntity.setProperty(DatastorePlaceGuideRepository.IMG_KEY_PROPERTY, IMG_KEY);
```

Comment:
This code is duplicated several times, extract it as a function.

Reply:
done